product_extended_segmentation/tests/test_extend_segmentation.py

```diff
@@ -0,0 +1,129 @@
+        cost_b = self.product_b_id.standard_price + \
+            tmpl_b_id.production_cost + \
+            tmpl_b_id.material_cost + tmpl_b_id.landed_cost + \
+            tmpl_b_id.subcontracting_cost
+
+        tmpl_d_id = self.product_d_id.product_tmpl_id
+        bom_d_id = tmpl_d_id.bom_ids[0]
+        wc_line_ids = bom_d_id.routing_id.workcenter_lines
+        routing_cost = sum([wcl_id.hour_nbr * wcl_id.workcenter_id.costs_hour
+                            for wcl_id in wc_line_ids])
+
+        self.assertTrue(routing_cost, "Routing Cost cannot be zero")
+
+        computed_cost = tmpl_d_id._calc_price(bom=bom_d_id)
+        self.assertEqual(computed_cost, cost_a + cost_b + routing_cost)
+
+    def compute_price_from_wizard(self, product_id=False, real_time=False,
+                                  recursive=False):
+        pid = False
+        if product_id._name == "product.template":
+            tmpl_id = product_id
+            pid = product.id
+        elif product_id._name == "product.product":
+            tmpl_id = product_id.product_tmpl_id
+            pid = tmpl_id.id
+
+        wizard_id = self.wizard.with_context({
+            'active_id': pid,
+            'active_model': 'product.template',
+            'active_ids': [pid]
+        }).create({
+            'real_time_accounting': real_time,
+            'recursive': recursive
+        })
+
+        wizard_id.compute_from_bom()
+        return True
+
+    def test_02_BoM_D_cost_compute(self):
+        tmpl_id = self.product_d_id.product_tmpl_id
+        before_compute = tmpl_id.standard_price
+        self.compute_price_from_wizard(self.product_d_id, False, True)
+
+        self.assertTrue(before_compute != tmpl_id.standard_price)
+
+        # Before compute cost: 50
+        # After compute cost : 35 ( material=30, production=5 )
+        self.assertTrue(before_compute - tmpl_id.standard_price == 15)
+
+    def test_03_BoM_E_cost_compute(self):
+        """
+        Compute Product(E) without using recursive
+        """
+        tmpl_id = self.product_e_id.product_tmpl_id
+        before_compute = tmpl_id.standard_price
+
+        self.compute_price_from_wizard(self.product_e_id, False, True)
+
+        # The cost changed
+        self.assertTrue(before_compute != tmpl_id.standard_price)
+
+        # Production cost should be: 15 = 10(E) + 5(D) production costs
+        self.assertTrue(tmpl_id.production_cost == 15)
+
+        # New Cost should be 75
+        self.assertTrue(tmpl_id.standard_price == 75)
+
+    def test_04_BoM_E_cost_compute_recursive(self):
+        """
+        Compute Product(E) using recursive calls
+        """
+        tmpl_d_id = self.product_d_id.product_tmpl_id
+        d_before_compute = tmpl_d_id.standard_price
+
+        self.compute_price_from_wizard(self.product_e_id, True, True)
+
+        # The cost for D-child have changed
+        self.assertTrue(d_before_compute != tmpl_d_id.standard_price)
```

Comment:
@osvalr Though you have explicitly assessed that STD has changed,
Assess too, the segmentation, providing which were the expected values to be assessed.
What were the values expected for M, L, P &S, and asserting against received values.